src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -345,12 +348,49 @@ worker_partial_agg_sfunc(PG_FUNCTION_ARGS)
 	FmgrInfo info;
 	int argumentIndex = 0;
 	bool initialCall = PG_ARGISNULL(0);
+	TupleDesc combinedInputDesc = NULL;
+	bool isNull;
 
 	if (initialCall)
 	{
+		if (PG_ARGISNULL(1))
+		{
+			ereport(ERROR, (errmsg("worker_partial_agg_sfunc received invalid null "
+								   "input for second argument")));
+		}
 		box = pallocInAggContext(fcinfo, sizeof(StypeBox));
 		box->agg = PG_GETARG_OID(1);
 
+		/* check if input comes combined into record */
+		if (RECORDOID == get_fn_expr_argtype(fcinfo->flinfo, 2))
+		{
+			if (PG_ARGISNULL(2))
+			{
+				ereport(ERROR, (errmsg(
+									"worker_partial_agg_sfunc received invalid input "
+									"for third argument: null record"),
+								errhint("Elements of record may be null")));
+			}
+
+			/* retrieve record description and store it in the box */
+			HeapTupleHeader record = PG_GETARG_HEAPTUPLEHEADER(2);
+
+			combinedInputDesc = lookup_rowtype_tupdesc(
+				HeapTupleHeaderGetTypeId(record),
+				HeapTupleHeaderGetTypMod(record)
+				);
+
+			box->combinedInputDesc = pallocInAggContext(fcinfo, TupleDescSize(
```

Comment:
With currently implemented flow this wouldn't work because we're performing type-checking (TypecheckWorkerPartialAggArgType) using an uninitialized box (before calling InitializeStypeBox)